### PR TITLE
Add a function for sourcing `profile` after crew install/reinstall/postinstall/upgrade

### PR DESCRIPTION
**Pull request by supechicken**

### CODE REVIEW

**Comment by uberhacker** [comment]
This won't work like you think it might.  The source is inside the bash script session only.  Also, by only including 4 crew commands, you lose all the others.

**Comment by supechicken** [comment]
> This won't work like you think it might. The source is inside the bash
script session only.

The `profile` file is `source` by `bashrc`, and all scripts under `*.d/`
will be `source` by `profile`, so all functions/variables will be exposed
to the shell

> Also, by only including 4 crew commands, you lose all the others.

You are right :)

Ed Reel ***@***.***> 於 2022年3月17日週四 下午9:58寫道：

**Comment by supechicken** [comment]
> This won't work like you think it might. The source is inside the bash
script session only.

The `profile` file is `source` by `bashrc`, and all scripts under `*.d/`
will be `source` by `profile`, so all functions/variables will be exposed
to the shell

> Also, by only including 4 crew commands, you lose all the others.

You are right :)

File: src/profile.d/crew.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function crew () {
+  case "${1}" in
+  'install'|'reinstall'|'postinstall'|'upgrade')
+    command crew "${@}" && source "${CREW_PREFIX}/etc/profile"
+  ;;
+  *)
+    command crew "${@}"
+  ;;
+  esac
+}
```
Comment (uberhacker) [inline src/profile.d/crew.sh:10]: I don't think this works like you think it might.

**Comment by Zopolis4** [comment]
Assuming this PR is actually possible (not sure), it would be great to see it cleaned up and merged-- we have 36 packages that manually ask the user to `source ~/.bashrc`, and it'd be nice to get rid of those lines.

**Comment by supechicken** [comment]
> Assuming this PR is actually possible (not sure), it would be great to see it cleaned up and merged-- we have 36 packages that manually ask the user to `source ~/.bashrc`, and it'd be nice to get rid of those lines.

Closing this as I have a better idea of achieving this...

**Comment by supechicken** [comment]
> Assuming this PR is actually possible (not sure), it would be great to see it cleaned up and merged-- we have 36 packages that manually ask the user to `source ~/.bashrc`, and it'd be nice to get rid of those lines.

Just comes out from my mind, we can use `SIGUSR1/2` to communicate between `crew` and the function for controlling  if `source .bashrc` is needed or not.

**Pull request closed without merging**